src/main/webapp/js/stats-page-loader.js

```diff
@@ -1,27 +1,27 @@
 // Fetch stats and display them in the page.
-  function fetchStats() {
-    const url = '/stats';
-    fetch(url).then((response) => {
-      return response.json();
-    }).then((stats) => {
-      const statsContainer = document.getElementById('stats-container');
-      statsContainer.innerHTML = '';
+function fetchStats() {
+  const url = '/stats';
+  fetch(url).then((response) => {
+    return response.json();
+  }).then((stats) => {
+    const statsContainer = document.getElementById('stats-container');
+    statsContainer.innerHTML = '';
 
-      const messageCountElement = buildStatElement('Message count: ' + stats.messageCount);
-      statsContainer.appendChild(messageCountElement);
+    const messageCountElement = buildStatElement('Message count: ' + stats.messageCount);
+    statsContainer.appendChild(messageCountElement);
 
-      const maxMsgLengthElement = buildStatElement('Maximum message length: ' + stats.maxMsgLength);
-      statsContainer.appendChild(maxMsgLengthElement);
-    });
-  }
+    const maxMsgLengthElement = buildStatElement('Maximum message length: ' + stats.maxMsgLength);
+    statsContainer.appendChild(maxMsgLengthElement);
+  });
+}
 
-    function buildStatElement(statString){
-    const statElement = document.createElement('p');
-    statElement.appendChild(document.createTextNode(statString));
-    return statElement;
-  }
+function buildStatElement(statString){
+  const statElement = document.createElement('p');
+  statElement.appendChild(document.createTextNode(statString));
+  return statElement;
+}
 
-  // Fetch data and populate the UI of the page.
-    function buildUI(){
-    fetchStats();
-  }
+// Fetch data and populate the UI of the page.
+function buildUI(){
```

Comment:
I wonder if this function is necessary? Since it just calls fetchStats()?

Reply:
I'm not super sure, I just asked @Meg1456 about it, she might be clearer about it coz she did the Stats guided project and to be honest I'm not very sure about the stuff in stats-page-loader.js file. I'll try to read through the guide later though after I'm done with my maps part.

Reply:
I think the function is not really necessary, it serves more like a wrapper. This function wrapping is everywhere in the codebase so that every html page basically contains a onload="buildUI()" even though the buildUI function only calls one other function. I think the point is either buildingUI may call multiple functions in the future or just to make all html pages consistent. I am also fine with just removing all buildUI functional wrappers if everyone agrees that less is more.

Reply:
Ah that makes sense. Yeah keeping buildUI makes sense then, as it keeps the style consistent among all the pages.

Reply:
Ok, sure